pkg/volume/emptydir/empty_dir.go

```diff
@@ -290,11 +290,11 @@ func (ed *emptyDir) setupHugepages(dir string) error {
 	}
 	// If the directory is a mountpoint with medium hugepages, there is no
 	// work to do since we are already in the desired state.
-	if isMnt && medium == v1.StorageMediumHugePages {
+	if isMnt && v1helper.IsHugePageMedium(medium) {
```

Comment:
is this correct? if you set up two volumes with different hugepages medium sizes, won't this short-circuit the second thinking it is already set up, when in fact it should fail because the same mount point can't be two different sizes?

previously, validation limited a pod to a single hugepages size

Reply:
I don't think this is the case here. Pod can't have two different medium sizes huge page mediums with the same mount path. For two different mount paths this check would serve its purpose: it will prevent attempts to mount already mounted medium to the same mount path again.

Reply:
previously, there could only be a single hugepages size per pod, so all hugepages mounts had to agree. with this PR, that is no longer the case, correct?

Reply:
Correct, but I'm not sure I follow.

From my understanding this check makes setupHugepages function idempotent by checking if volume is already mounted.

Reply:
previously there were two aspects to be idempotent:
* is it mounted
* is it hugepages

there are now three aspects:
* is it mounted
* is it hugepages
* is it the right size

I don't see the size aspect represented

Reply:
Got it eventually. Thank you for the explanations! I'll update the PR soon.

Reply:
Just to confirm before implementing it. I'm going to get page size for hugetlb mounts using k8s.io/utils/mount API and compare it to the emptyDir medium size. If they don't match setupHugepages should fail.

Reply:
I will defer to sig-storage or sig-node folks ... this PR introduces the possibility of this conflict, so I'm not sure what the correct resolution is.

Reply:
@liggitt @derekwaynecarr I've implemented additional check in GetMountMedium function. It checks if mounted pagesize is the same as medium size. Please review. I put this change into a separate commit that it's easier to review.

Reply:
@liggitt could you please give more details about the conflict you see here? Thanks!

Reply:
> previously, there could only be a single hugepages size per pod, so all hugepages mounts had to agree. with this PR, that is no longer the case, correct?

Reply:
with the additional checking added by @bart0sh, is it still an issue? 
Maybe we could add feature gate around this code? If feature is enabled, it could support different sizes?

Reply:
@jingxu97 Thank you for the review!

Feature gate has been already added, I believe. It's in the first commit of this PR. It's disabled by default, i.e. pods that use hugepages-<size> syntax will not pass validation.

I've updated the PR so that page size is obtained from volume mount options only if requested medium uses hugepages-<size> syntax. Please have a look.